### PR TITLE
[WiP] GTK: add OpenGL ES support

### PR DESCRIPTION
**Pull request by thesourcehim**
Added support for GLES in posix/gtk frontend. Does not work for now, initialization fails with missing extension. See #812.

### CODE REVIEW

**Comment by thesourcehim** [comment]
Figured what was wrong with initialization. Starts properly now, but the image renders incorrectly, like it is partially visible behind black surface

gles_test

File: desmume/src/frontend/posix/gtk/graphics.ui [hunk 1]
```diff
@@ -65,6 +65,7 @@
                   <item translatable="yes">Null</item>
                   <item translatable="yes">SoftRasterizer</item>
                   <item translatable="yes">OpenGL</item>
+				  <item translatable="yes">OpenGL ES</item>
```
Comment (rofl0r) [inline desmume/src/frontend/posix/gtk/graphics.ui:68]: you seem to have mixed tabs and spaces here

Comment (thesourcehim) [reply desmume/src/frontend/posix/gtk/graphics.ui]: Fixed

**Comment by thesourcehim** [comment]
Setting multisampling to something other than 'None' makes problem less apparent. Also 2d renders fine it looks like, it's only 3d elements that hide behind blackness.

**Comment by rogerman** [comment]
Sorry, but there are some fundamental problems with the OpenGL ES renderer at the moment, which is why no port was updated to use it just yet. But I'm getting there. Just wait a few days and check back.

**Comment by thesourcehim** [comment]
I just tested with commit 60385bd and the situation has improved somewhat: less black tint on nvidia gpu, but still present
nvidia
And it seems that the higher the fps the more black tint is on screen (threading issue?)

On intel gpu some blue dots appeared over blackness, but overall everything is still black on intel.

**Comment by rogerman** [comment]
Thanks for checking, but the ES renderer isn't meant to be working 100% just yet. It's still in a testing phase. As of now, there is still one major fundamental problem that exists.

When you see the "egl_3Demu.cpp" and "egl_3Demu.h" files appear under gtk2, then you'll know that the OpenGL ES renderer is "done".

**Comment by rogerman** [comment]
The ES renderer is now complete as of commit be51e41. I've updated all of the POSIX frontends to work with the new ES renderer using some of your work as inspiration.

- If you're building with Autotools, you'll need to use --enable-opengles
- If you're building with Meson, you'll need to set -Dopengles=true

**Pull request closed without merging**